msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -22,73 +22,102 @@
 //  THE SOFTWARE.
 package com.microsoft.identity.client;
 
-/**
- * MSAL API surface implementation of an account.
- */
-class Account implements IAccount {
-
-    private IAccountIdentifier mAccountIdentifier;
-    private IAccountIdentifier mHomeAccountIdentifier;
-    private String mUsername;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.microsoft.identity.common.internal.logging.Logger;
+import com.microsoft.identity.common.internal.providers.microsoft.MicrosoftIdToken;
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+
+import java.text.ParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Account implements IAccount {
+
+    private static final String TAG = Account.class.getSimpleName();
+
+    private final String mRawIdToken;
+    private String mHomeOid;
+    private String mHomeTenantId;
     private String mEnvironment;
 
-    Account() {
-        // Empty constructor
+    public Account(@Nullable final String rawIdToken) {
+        mRawIdToken = rawIdToken;
     }
 
-    /**
-     * Sets the account id.
-     *
-     * @param accountId The IAccountIdentifier to set.
-     */
-    void setAccountIdentifier(final IAccountIdentifier accountId) {
-        mAccountIdentifier = accountId;
+    void setId(@Nullable final String id) {
+        mHomeOid = id;
     }
 
+    @NonNull
     @Override
-    public IAccountIdentifier getAccountIdentifier() {
-        return mAccountIdentifier;
-    }
+    public String getId() {
+        String id;
+
+        if (null != mRawIdToken && null != getClaims()) {
+            id = (String) getClaims().get(MicrosoftIdToken.OBJECT_ID);
+        } else {
+            id = mHomeOid;
+        }
 
-    /**
-     * Sets the home account id.
-     *
-     * @param homeAccountId The IAccountIdentifier to set.
-     */
-    void setHomeAccountIdentifier(final IAccountIdentifier homeAccountId) {
-        mHomeAccountIdentifier = homeAccountId;
+        return id;
     }
 
-    @Override
-    public IAccountIdentifier getHomeAccountIdentifier() {
-        return mHomeAccountIdentifier;
+    void setTenantId(@Nullable final String tenantId) {
+        mHomeTenantId = tenantId;
     }
 
-    /**
-     * Sets the username.
-     *
-     * @param username The username to set.
-     */
-    void setUsername(final String username) {
-        mUsername = username;
+    @Nullable
+    String getTenantId() {
+        return mHomeTenantId;
     }
 
-    @Override
-    public String getUsername() {
-        return mUsername;
+    @Nullable
+    String getHomeAccountId() {
+        return getId() + "." + mHomeTenantId;
```

Comment:
We want to use the objectid of the home tenant as well.

Reply:
Yep - this is already implemented [here](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/iambmelt/tenant-profile-redux/msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java#L212-L215)